first_order_model/shrink_util.py

```diff
@@ -951,82 +1014,83 @@ def compute_deletion(layer_graph,
-    # SO copy whatever we do here to the other operand
+    # So copy whatever we do here to the other operand
+    # A mirror means that two layers outputs get treated the same, so if index 1 is deleted in one,
+    # it will be deleted in the other.
     for mirror in layer_graph[layer].mirror:
-        # Start a new custom deletion for the mirror
+        # Start a new custom deletion for the mirror. The only other reason is tie, which is treated a little differently.
         if reason != 'mirror':
             custom_deletions.append((mirror, copy.copy(deletions[layer]), 'mirror'))
             print("Starting a mirrorred deletion")
 
-    # Figure out what you need to delete for the following layers
+    # Figure out what you need to delete from the inputs of the following convs, and the relevant portions of batchnorms.
+    # Following layers contains any layeres who can be affected by the deletion in layer_graph[layer]
     for following_layer in following_layers:
+        # If we have already addressed this layer, continue
         if following_layer in deletions:
             continue
 
         deletions[following_layer] = []
+        
+        # Counts up for each index in the inputs for this layer
         counter = 0
         for previous_layer in layer_graph[following_layer].before:
+
+            # If you don't delete from the previous layer, just increment the counter by its ouputs.
             if previous_layer not in deletions:
-                counter += layer_graph[previous_layer].o
+                counter += layer_graph[previous_layer].out_channels
                 continue
 
-            # If delete from previous layer, delete corresponding element from this layer
+            # If delete from previous layer, delete corresponding element from this layer and increment the counter
             for deletion in deletions[previous_layer]:
                 deletions[following_layer].append(
                     (counter + deletion[0], counter + deletion[1]))
-            counter += layer_graph[previous_layer].o
+            counter += layer_graph[previous_layer].out_channels
 
-        # If there is another tied after layer i.e. a layer who has their output tied to the input of this layer, like in resnet, we need to delete from their outputs as well
+        # If there is another tied after layer i.e. a layer who has their output tied to the input of this layer, 
+        # like in resnet, we need to delete from their outputs as well
         if len(layer_graph[following_layer].tied_after) != 0:
-
-            # Calculate the number of elements to delete
-            total_deleted = 0
-            for del_tuple in deletions[following_layer]:
-                total_deleted += del_tuple[1] - del_tuple[0]
+            total_deleted = copy.copy(deletions[following_layer])
 
             # Add the output layer to our work queue (custom deletions)
             for after_tied in layer_graph[following_layer].tied_after:
```

Comment:
these variable names are confusing, please rename them

Reply:
Should be better now.